bundler/lib/bundler/errors.rb

```diff
@@ -246,4 +246,17 @@ def message
   end
 
   class InvalidArgumentError < BundlerError; status_code(40); end
+
+  class PsychLinkedToIncompatibleOrMissingRubyError < BundlerError
+    def initialize(orig_exception)
+      full_message = "The psych library used by RubyGems to load its configuration is linked to an incompatible or missing Ruby.\n" \
+                     "The underlying error was #{orig_exception.class}: #{orig_exception.message}, with backtrace:\n" \
```

Comment:
Should we use detailed_message to show the cause instead?

Reply:
I didn't know about `Exception#detailed_message`! Here I copied the approach used by other custom exceptions but perhaps we can consider migrating all of them to use `Exception#detailed_message` at once? I can reticket that.

Reply:
I opened https://github.com/rubygems/rubygems/issues/8067 for this!